### DISCUSSION TITLE
Prolog is being detected as Raku

### DISCUSSION THREAD

**Discussion by cbarrick** [github-linguist/linguist#6454] [Classification] [answered]
Example repo: https://github.com/cbarrick/practice

Prolog and Perl notoriously use the same file extension: `.pl`. IIUC Raku developed from Perl. I assume this is somehow related to the misclassification.

The issues page told me to open a discussion before opening an issue, so here I am.

**Comment by lildude** [comment, marked as answer]
This is happening because one massive file is being detected as Raku and not Prolog because it's content doesn't match any of the [heuristics for the `.pl` extension](https://github.com/github-linguist/linguist/blob/d352058f979caab69b5c0876cd4992ce43a29132/lib/linguist/heuristics.yml#L511-L520) so it's falling through to the classifier which is not making a good guess, which isn't surprising given the content:

https://github.com/cbarrick/practice/blob/master/reddit/dailyprogrammer/246-med/words.pl

It's 3.64 MB so is by far the largest by bytes of code so if having a dispropotionate effect on the stats.

You'll need to implement an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md) to either force the language, or ignore the file.

**Reply by cbarrick** [reply]
Thanks!
